### PR TITLE
Query components without refcell

### PR DESCRIPTION
**Pull request by bigfoodK**
# What I did
## Query components without refcell
Also, there is no entity returned.
### Before
```rust
let (entity: Entity, (a: Ref<A>, b: Ref<B>, c: Ref<C>)) = app.query<&A, &B, &C>();
```
### After
```rust
let (a: &A, b: &B, c: &C) = app.query<A, B, C>();
```

## Support optional component
```rust
let (a: &A, c: Option<&C>) = app.query<A, Option<C>>();
```

### CODE REVIEW

**Comment by namse** [comment]
> Also, there is no entity returned.

Why did you decide not to return entity?

**Comment by bigfoodK** [comment]
From now on, the component belongs to the entity. So if we borrow component as mut, then we can not return entity together.

File: luda-adventure/src/ecs/component.rs [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+    }
+    pub fn as_ref_mut(&mut self) -> &mut T {
+        &mut self.component
+    }
+}
+
+pub trait ContainedComponent: Debug + 'static {
+    fn as_any(&self) -> &dyn std::any::Any;
+    fn as_any_mut(&mut self) -> &mut dyn std::any::Any;
+}
```
Comment (namse) [inline luda-adventure/src/ecs/component.rs:28]: Why did you need `ComponentContainer` and `ContainedComponent`, not only just `Component`?

**Comment by namse** [comment]
> From now on, the component belongs to the entity. So if we borrow component as mut, then we can not return entity together.

Then how user can determine which entity queried?

File: luda-adventure/src/ecs/component.rs (hunk 1 above)
Comment (bigfoodK) [reply luda-adventure/src/ecs/component.rs:28]: For downcasting. If we store them as `Vec<Box<dyn component>>`, It's typeid will be `Box<dyn component>`'s, not specific component's like `Box<Mover>`.

**Comment by bigfoodK** [comment]
> > From now on, the component belongs to the entity. So if we borrow component as mut, then we can not return entity together.
> 
> Then how user can determine which entity queried?
User may not know.

**Comment by namse** [comment]
@bigfoodK Let's make sure what this PR want to do.
No RefCell used in last code.
What's the purpose of this PR?

**Comment by bigfoodK** [comment]
Deparated
